sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/NpmVersionHelper.java

```diff
@@ -37,18 +35,38 @@ public class NpmVersionHelper {
 	public String getResolvedVersionFromNpmRegistry(PackageJson packageJson, String versionRange) {
 		try {
 			String packageName = packageJson.getName();
-			JsonNode jsonArray = getPackageMetadata(versionRange, packageName);
-			if (jsonArray.isEmpty()) {
-				throw new RuntimeException("No versions found for the specified range: " + versionRange);
+			JsonNode metadata = getPackageMetadata(packageName, versionRange);
+			if (metadata.isEmpty()) {
+				log.warn("No versions found for range: {}. Skipping.", versionRange);
+				return null;
 			}
-			
-			JsonNode jsonObject = jsonArray.get(0);
-			return jsonObject.get("version").asText();
+
+			return metadata.get(0).get("version").asText();
+		} catch (IOException | InterruptedException e) {
+			log.error("Error retrieving resolved version: {}", e.getMessage());
+			throw new RuntimeException("Error resolving version", e);
+		}
+	}
+
+	private static JsonNode getPackageMetadata(String packageName, String versionRange) throws IOException, InterruptedException {
+		if (packageName == null || packageName.isEmpty()) {
+			throw new IllegalArgumentException("Package name cannot be null or empty");
+		}
+
+		ProcessBuilder processBuilder = new ProcessBuilder().command("npm", "pack", "--dry-run", "--json", packageName + "@" + versionRange)
+				.redirectErrorStream(true);
+		Process process = processBuilder.start();
+		StringBuilder output = new StringBuilder();
+
+		try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8))) {
+			reader.lines().forEach(output::append);
 		}
```

Comment:
FWIW, I'd probably use IOUtils.toString() over either this version or the previous version. Two reasons:

1. Like the original version, it mostly works on bytes rather than strings.
2. I've run into some funky cases with `BufferedReader` where no newline character is output and the output starts exceeding the buffer's capacity. Maybe that's been fixed though?